packages/kbn-esql-validation-autocomplete/src/validation/resources.ts

```diff
@@ -29,6 +29,17 @@ export async function retrieveFields(
   if (!callbacks || commands.length < 1) {
     return new Map();
   }
+  // Do not fetch fields, if query has only one source command and that command
+  // does not require fields.
+  if (commands.length === 1) {
+    switch (commands[0].name) {
+      case 'from':
```

Comment:
I think you can drop the `row` command as it is already handled by the next condition. No matter ast length.

Also, this could be reduced into:
```ts
if(commands.length === 1 && ["from", "show"].includes(commands[0].name)){
  return new Map();
}